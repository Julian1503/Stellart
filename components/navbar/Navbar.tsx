import Image from "next/image";
import Link from "next/link";
import classes from "./navbar.module.css";
import { Aboreto } from '@next/font/google';


const aboretoFont = Aboreto({weight:"400", style:"normal", subsets:["latin"]});

function Navbar() {
  return (
    <nav className={classes.navbar}>
      <div className={classes.image_container}>
        <Image src="/images/stellart.svg" alt="Logo" width={250} height={100}/>
      </div>
      <div className={classes.links_container}>
        <ul>
          <li className={classes.link}>
            <Link className={aboretoFont.className} href="">Artwork</Link>
          </li>
          <li className={classes.link}>
            <Link className={aboretoFont.className} href="">Artists</Link>
          </li>
          <li className={classes.link}>
            <Link className={aboretoFont.className} href="">Expositions</Link>
          </li>
          <li className={classes.link}>
            <Link className={aboretoFont.className} href="">Login</Link>
          </li>
        </ul>
      </div>
    </nav>
  );
}

export default Navbar;
