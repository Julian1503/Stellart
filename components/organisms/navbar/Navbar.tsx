import Image from "next/image";
import classes from "./navbar.module.css";
import { useState } from "react";
import { NavLink } from "@/constants/types";
import Overlay from "@/atoms/overlay/Overlay";
import NavbarMenu from "@/molecules/navbar-menu/NavbarMenu";

type NavbarProps = {
  links: Array<NavLink>;
  iconSrc: string;
};

function Navbar({ links, iconSrc }: NavbarProps) {
  const [isMenuOpen, setMenuOpen] = useState<boolean>(false);
  return (
    <nav data-testid="navbar" className={classes.navbar}>
      <div data-testid="logo-container" className={classes.image_container}>
        <Image data-testid="logo" src={iconSrc} alt="Logo" width={190} height={100} />
      </div>
      <NavbarMenu showLoginButton links={links} isMenuOpen={isMenuOpen} handleMenu={ () => setMenuOpen((prevState) => !prevState)}/>
      <Overlay 
        onClick={() => setMenuOpen(false)}
        showOverlay={isMenuOpen}
      />
    </nav>
  );
}

export default Navbar;
