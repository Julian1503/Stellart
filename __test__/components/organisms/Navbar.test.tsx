import React from "react";
import { render, fireEvent, cleanup, waitFor } from "@testing-library/react";
import Navbar from "@/components/organisms/navbar/Navbar";

const links = [
  { id: 1, link: "/home", text: "Home" },
  { id: 2, link: "/about", text: "About" },
  { id: 3, link: "/contact", text: "Contact" },
];

afterEach(cleanup);
const ICON_SRC = "/images/stellart.svg";

describe("Navbar component", () => {
  it("renders without crashing", () => {
    const { getByTestId } = render(<Navbar links={links} iconSrc={ICON_SRC} />);
    const navbar = getByTestId("navbar");
    expect(navbar).toBeInTheDocument();
  });

  it("renders the logo with the correct src", () => {
    const { getByTestId } = render(<Navbar links={links} iconSrc={ICON_SRC} />);
    const logo = getByTestId("logo");
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute("src", ICON_SRC);
  });

  it("renders the menu with the correct links", () => {
    const { getByTestId } = render(<Navbar links={links} iconSrc={ICON_SRC} />);
    const menu = getByTestId("menu");
    expect(menu).toBeInTheDocument();
    links.forEach((link) => {
      const navButton = getByTestId(`btnNav${link.id}`);
      expect(navButton).toBeInTheDocument();
      expect(navButton).toHaveTextContent(link.text);
    });
  });

  it("opens and closes the menu when the burger button is clicked", async () => {
    const { getByTestId } = render(<Navbar links={links} iconSrc={ICON_SRC} />);
    const btnBurger = getByTestId("btnBurger");
    const btnClose = getByTestId("btnClose");
    const overlay = getByTestId("overlay");

    expect(btnClose).toHaveClass("btn_close_menu_closed");
    expect(btnClose).not.toHaveClass("btn_close_menu_opened");
    expect(overlay).not.toHaveClass("active");

    fireEvent.click(btnBurger);

    await waitFor(() => {
      expect(btnClose).not.toHaveClass("btn_close_menu_closed");
      expect(btnClose).toHaveClass("btn_close_menu_opened");
      expect(overlay).toHaveClass("active");
    });

    fireEvent.click(btnClose);

    await waitFor(() => {
      expect(btnClose).toHaveClass("btn_close_menu_closed");
      expect(btnClose).not.toHaveClass("btn_close_menu_opened");
      expect(overlay).not.toHaveClass("active");
    });
  });

  it("matches snapshot", () => {
    const button = render(<Navbar links={links} iconSrc={ICON_SRC} />);
    expect(button).toMatchSnapshot();
  });
});
