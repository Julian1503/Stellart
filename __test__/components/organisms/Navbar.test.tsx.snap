// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar"
        data-testid="navbar"
      >
        <div
          class="image_container"
          data-testid="logo-container"
        >
          <img
            alt="Logo"
            data-nimg="1"
            data-testid="logo"
            decoding="async"
            height="100"
            loading="lazy"
            src="/images/stellart.svg"
            style="color: transparent;"
            width="190"
          />
        </div>
        <span
          arial-label="Burger button"
          class="menu_btn btn_burger"
          data-testid="btnBurger"
        >
          <span
            class="burger"
          >
            <span
              class="line"
            />
            <span
              class="line"
            />
            <span
              class="line"
            />
          </span>
        </span>
        <div
          class="links_container  "
          data-testid="menu"
        >
          <span
            class="btn_close btn_close_menu_closed"
            data-testid="btnClose"
          >
            <span
              class="line"
            />
            <span
              class="line"
            />
          </span>
          <ul>
            <li
              class="link "
              data-testid="btnNav1"
            >
              <a
                href="/home"
              >
                Home
              </a>
            </li>
            <li
              class="link "
              data-testid="btnNav2"
            >
              <a
                href="/about"
              >
                About
              </a>
            </li>
            <li
              class="link "
              data-testid="btnNav3"
            >
              <a
                href="/contact"
              >
                Contact
              </a>
            </li>
            <li>
              <a
                arial-name="Button"
                class="btn shadow undefined"
                data-testid="btn"
                href="/Login"
                style="width: 170px; height: 49px;"
              >
                Login
              </a>
            </li>
          </ul>
        </div>
        <div
          class=" overlay   "
          data-testid="overlay"
        />
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar"
      data-testid="navbar"
    >
      <div
        class="image_container"
        data-testid="logo-container"
      >
        <img
          alt="Logo"
          data-nimg="1"
          data-testid="logo"
          decoding="async"
          height="100"
          loading="lazy"
          src="/images/stellart.svg"
          style="color: transparent;"
          width="190"
        />
      </div>
      <span
        arial-label="Burger button"
        class="menu_btn btn_burger"
        data-testid="btnBurger"
      >
        <span
          class="burger"
        >
          <span
            class="line"
          />
          <span
            class="line"
          />
          <span
            class="line"
          />
        </span>
      </span>
      <div
        class="links_container  "
        data-testid="menu"
      >
        <span
          class="btn_close btn_close_menu_closed"
          data-testid="btnClose"
        >
          <span
            class="line"
          />
          <span
            class="line"
          />
        </span>
        <ul>
          <li
            class="link "
            data-testid="btnNav1"
          >
            <a
              href="/home"
            >
              Home
            </a>
          </li>
          <li
            class="link "
            data-testid="btnNav2"
          >
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li
            class="link "
            data-testid="btnNav3"
          >
            <a
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li>
            <a
              arial-name="Button"
              class="btn shadow undefined"
              data-testid="btn"
              href="/Login"
              style="width: 170px; height: 49px;"
            >
              Login
            </a>
          </li>
        </ul>
      </div>
      <div
        class=" overlay   "
        data-testid="overlay"
      />
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
